### ISSUE TITLE
BLE ASSERT_PARAM crash when scanning takes too long

### ISSUE THREAD

**Issue by ThePizzaguy** [nkolban/esp32-snippets#352]
Hi Kolban,

Recently working with your code on a project I get this error if i use my code in the onResult function from your BLEScan. 
It only happens when my code takes a certain amount of time because if I use alot simpler code it works fine for hours.
![image](https://user-images.githubusercontent.com/9423611/34666166-469feb38-f463-11e7-8657-830828dd17fd.png)

I did see your post on the ESP32 forums about this from a month ago and were wondering if you found a fix.

Best Regards

Oscar

**Comment by chegewara** [comment]
If you are working on esp-idf then you can try one thing. In menuconfig find watchdog time and try to increase it. Im not saying it will help, but from screenshot we can see it may be related.

**Comment by ThePizzaguy** [comment]
Yes i am using esp-idf and that gave some interesting results, the code lasted somewhat longer and the error message was longer as well but sadly it didn't work. I found that the length of the error message corresponds with the time it takes in the function.

![image](https://user-images.githubusercontent.com/9423611/34667133-f76bbaa6-f467-11e7-95ac-df8981143933.png)
